filip/semantics/semantic_models.py

```diff
@@ -0,0 +1,1683 @@
+        return attr
+
+    def get_all_field_names(self) -> List[str]:
+        """
+        Get all field names which this property creates in the fiware
+        instance
+        """
+        pass
+
+    class Config:
+        underscore_attrs_are_private = True
+
+
+class Command(DeviceProperty):
+    """
+    Model describing a command property of an IoT device.
+
+    The command will add three fields to the fiware instance:
+        - name - Used to execute the command, function: send()
+        - name_info - Used to retrieve the command result: get_info()
+        - name_status - Used to see the current status: get_status()
+
+    A command can only belong to one field of one instance. Assigning it to
+    multiple fields will result in an error.
+    """
+
+    def send(self):
+        """
+        Execute the command on the IoT device
+
+        Raises:
+            Exception: If the command was not yet saved to Fiware
+        """
+        client = self._instance_link.semantic_manager.get_client(
+                 self._instance_link.instance_identifier.header)
+
+        context_command = NamedCommand(name=self.name, value="")
+        identifier = self._instance_link.instance_identifier
+        client.post_command(entity_id=identifier.id,
+                            entity_type=identifier.type,
+                            command=context_command)
+        client.close()
+
+    def get_info(self) -> str:
+        """
+        Retrieve the executed command result from the IoT-Device
+
+        Raises:
+            Exception: If the command was not yet saved to Fiware
+        """
+        return self._get_field_from_fiware(field_name=f'{self.name}_info',
+                                           required_type="commandResult").value
+
+    def get_status(self):
+        """
+        Retrieve the executed command status from the IoT-Device
+
+        Raises:
+            Exception: If the command was not yet saved to Fiware
+        """
+        return self._get_field_from_fiware(field_name=f'{self.name}_status',
+                                           required_type="commandStatus").value
+
+    def get_all_field_names(self) -> List[str]:
+        """
+        Get all the field names that this command will add to Fiware
+        """
+        return [self.name, f"{self.name}_info", f"{self.name}_result"]
+
+    class Config:
+        """if the name is changed the attribute needs to be removed
+        and re-added to the device. With frozen that logic is more clearly
+        given in the library. Further it allows us to hash the object"""
+        frozen = True
+
+
+class DeviceAttributeType(str, Enum):
+    """
+    Retrieval type of the DeviceAttribute value from the IoT Device into Fiware
+    """
+    lazy = "lazy"
+    """The value is only read out if it is requested"""
+    active = "active"
+    """The value is kept up-to-date"""
+
+
+class DeviceAttribute(DeviceProperty):
```

Comment:
why issn#t this mdoel in iot?

Reply:
It is a higher order abstraction of the object in IoT models.